code/controllers/subsystem/ambience.dm

```diff
@@ -29,7 +29,7 @@ SUBSYSTEM_DEF(ambience)
 		if (C)
 			var/mob/M = C.mob
 
-			if(M)
+			if(M && M.client)
```

Comment:
Wait... So you check the client exists, get the client's mob, check that the mob exists... But you need to check that the client exists again? How could it not exist?

Reply:
You're asking me

Reply:
I assume If the server is lagging somewhat, then the list from the previous run will be carried over to the next to continue processing but doesn't account for client disconnects. So if a client disconnects between fires, the list will still contain that client and their mob, however the mob will have no client attached to it because the client disconnected.